lib/api/video.js

```diff
@@ -200,6 +200,36 @@ class Video extends ApiResource {
         return Promise.resolve({channelId: channelId, videoId: videoId})
       })
   }
+
+  /**
+   * List all video metadata values
+   *
+   * @param {Number} videoId - ID of existing video.
+   */
+  listMetadata (videoId) {
+    return this.context.authRequest('get', `/videos/${videoId}/custom-metadata.json`)
+      .then((res) => {
+        return Promise.resolve(res.metadata)
+      })
+  }
+
+  /**
+   * Set video metadata value
+   * @param {Number} videoId - ID of existing video.
+   * @param {Number} fieldId - ID of existing metadata field.
+   * @param {*} value - Data to be stored in metadata field.
+   */
+  setMetadata (videoId, fieldId, value) {
+    let payload = qs.stringify({
+      value
+    })
+
+    return this.context.authRequest('put',
+        `/videos/${videoId}/custom-metadata/${fieldId}.json`, payload)
+      .then((res) => {
```

Comment:
Since this method is resolving the immediate value of the promise, ` return Promise.resolve(res)` can be removed. The following is equivalent to what you have here.

```
return this.context.authRequest('put', `/videos/${videoId}/custom-metadata/${fieldId}.json`, payload)
```